### PR TITLE
Unify loggers across application

### PR DESCRIPTION
**Pull request by hsienchiaolee**
## Issue
#1174 

Note:
* Purposely chunked changes into different commits per package.
• There are still some loggers that may need to be replaced. Skipped in this PR given lack of context and want to avoid breaking things unintentionally.

## Test
N/A, ran `./test/test_before_submit.sh`

## TODO
N/A

### CODE REVIEW

**Comment by harmony-ek** [comment]
Jenkins speaking; can one of the admins verify this patch?

**Comment by LeoHChen** [comment]
Jenkins, test this please.

File: api/proto/node/node.go [hunk 1]
```diff
@@ -116,7 +116,8 @@ func DeserializeBlockchainSyncMessage(d []byte) (*BlockchainSyncMessage, error)
 	decoder := gob.NewDecoder(bytes.NewReader(d))
 	err := decoder.Decode(&blockchainSyncMessage)
 	if err != nil {
-		log.Panic(err)
+		utils.GetLogger().Crit("Error", err)
+		panic(err)
```
Comment (LeoHChen) [inline api/proto/node/node.go]: is log.Panic will panic the process?  Or do we need to panic it here?  @mikedoan please review.

File: internal/utils/utils.go [hunk 2]
```diff
@@ -64,7 +63,8 @@ func ConvertFixedDataIntoByteArray(data interface{}) []byte {
 	buff := new(bytes.Buffer)
 	err := binary.Write(buff, binary.BigEndian, data)
 	if err != nil {
-		log.Panic(err)
+		GetLogger().Crit("Failed to convert fixed data into byte array", "err", err)
+		panic(err)
```
Comment (LeoHChen) [inline internal/utils/utils.go]: don't think we need to panic here.  @mikedoan

File: internal/utils/utils.go [hunk 3]
```diff
@@ -73,7 +73,8 @@ func ConvertFixedDataIntoByteArray(data interface{}) []byte {
 func GetUniqueIDFromIPPort(ip, port string) uint32 {
 	reg, err := regexp.Compile("[^0-9]+")
 	if err != nil {
-		log.Panic("Regex Compilation Failed", "err", err)
+		GetLogger().Crit("Regex Compilation Failed", "err", err)
+		panic(err)
```
Comment (LeoHChen) [inline internal/utils/utils.go]: same thing, no need to panic here.

File: internal/utils/utils.go [hunk 4]
```diff
@@ -199,15 +200,16 @@ func LoadKeyFromFile(keyfile string) (key p2p_crypto.PrivKey, pk p2p_crypto.PubK
...
 		key, pk, err = GenKeyP2PRand()
 		if err != nil {
-			log.Panic("LoadKeyFromFile", "GenKeyP2PRand Error", err)
+			GetLogger().Crit("LoadKeyFromFile", "GenKeyP2PRand Error", err)
+			panic(err)
 		}
 		err = SaveKeyToFile(keyfile, key)
 		if err != nil {
-			log.Print("LoadKeyFromFile", "failed to save key to keyfile", err)
+			GetLogger().Info("failed to save key to keyfile", "keyfile", err)
```
Comment (LeoHChen) [inline internal/utils/utils.go]: this is error

File: api/proto/node/node.go (hunk 1 above)
Comment (hsienchiaolee) [reply api/proto/node/node.go]: [GoDoc](https://golang.org/pkg/log/#Panic)
> Panic is equivalent to Print() followed by a call to panic().

File: internal/utils/utils.go (hunk 3 above)
Comment (hsienchiaolee) [reply internal/utils/utils.go]: If we are certain that this regex is always going to compile, can we skip the error check completely?

File: internal/utils/utils.go (hunk 2 above)
Comment (hsienchiaolee) [reply internal/utils/utils.go]: Not sure about the context here, but what's the damage if it fails to write?

File: node/contract.go [hunk 5]
```diff
@@ -148,7 +147,7 @@ func (node *Node) GetNonceOfAddress(address common.Address) uint64 {
 func (node *Node) GetBalanceOfAddress(address common.Address) (*big.Int, error) {
 	state, err := node.Blockchain().State()
 	if err != nil {
-		log.Error("Failed to get chain state", "Error", err)
+		utils.GetLogger().Error("Failed to get chain state", "Error", err)
```
Comment (mikedoan) [inline node/contract.go]: GetLogger is an expensive call - we don't have to it extensively

Comment (mikedoan) [reply node/contract.go]: @harmony-ek it was EK opinion. Please talk with him.

File: api/proto/node/node.go (hunk 1 above)
Comment (LeoHChen) [reply api/proto/node/node.go]: @mikedoan , do we need to panic here? please comment.

File: internal/utils/utils.go (hunk 3 above)
Comment (LeoHChen) [reply internal/utils/utils.go]: agree, not need to check the error here with the simple regex.

File: api/proto/node/node.go (hunk 1 above)
Comment (rlan35) [reply api/proto/node/node.go]: Don't panic here please. We should avoid panic (exit the software) as much as possible

File: core/blockchain.go [hunk 6]
```diff
@@ -1123,7 +1122,7 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 	for i, block := range chain {
 		// If the chain is terminating, stop processing blocks
 		if atomic.LoadInt32(&bc.procInterrupt) == 1 {
-			log.Debug("Premature abort during blocks processing")
+			utils.GetLogger().Debug("Premature abort during blocks processing")
```
Comment (rlan35) [inline core/blockchain.go]: These change on ethereum's log code will potentially add a lot more logs for us. Can you test the logs before and after this change and see how much difference in the size?

Comment (hsienchiaolee) [reply core/blockchain.go]: Ran both versions for 10 minutes, barely any increase:
• Before: ~2.0M
• After: ~2.1M
Seems like most of these are not called or are `Error` log types (which shouldn't happen often).

**Pull request merged**